### PR TITLE
Define new manual release strategy

### PR DESCRIPTION
**Pull request by paleloser**
With this PR, a new `release` workflow is introduced. This workflow:
* Bumps the version of the `package.json` and `package-lock.json` to the one passed
* Tags the released commit with the given version
* Rolls the major tag associated to the version to the released commit

With the creation of this workflow, I've removed the version-per-commit we used to have. This way we have more control over the action versioning, and allow users to:
* Stay synced with the `main` branch (`@main`)
* Select a specific major train (`@v1`)
* Pin a specific release (`@v1.2.3`)

Note that this is nothing related to GitHub Releases, which we would need to manually generate after having released a version. In the end GitHub Releases is purely informational (docs).

**TBD**
- [x] Update `CONTRIBUTING.md` with the new releasing procedure

### CODE REVIEW

**Review by gdelgadot — APPROVED** [review]

**Review by beltran-rubo — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          node-version: 16.x
+
+      - name: Git config
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+      - name: Release version
+        run: npm version ${{ inputs.version }} -m "[AUTOMATED] Release version %s"
```
Comment (mpermar) [inline .github/workflows/release.yaml:31]: Do you need a write access token for doing this? I think in the previous workflow that was the point of having secrets.VIB_ACTION_TOKEN

**Comment by mpermar** [comment]
I like the automated process more but I'm totally fine with this.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (paleloser) [reply .github/workflows/release.yaml:31]: Uhmm, I [tested](https://github.com/paleloser/vmware-image-builder-action/actions/runs/4303215635/jobs/7502671830) this in my own fork and worked, so I guess it should pass here too. Note that I added `contents: write` on the workflow `permissions`.

Comment (paleloser) [reply .github/workflows/release.yaml]: The `--allow-same-version` seems useful, added in [75240d2](https://github.com/vmware-labs/vmware-image-builder-action/pull/140/commits/75240d2285a50dab8c42221a6ebef0b2d86dd9c8), thanks!

**Pull request merged**